### DISCUSSION TITLE
changing bibtex-completion-bibliography on a per-document basis

### DISCUSSION THREAD

**Discussion by ramos** [emacs-citar/citar#148] [Ideas]
Sometimes, specially when working collaboratively in a paper, it is useful to have the .bib file of the current document (that sometimes is a local file and not your bib database) added to the bibtex-completion-bibliography.

**Comment by bdarcus** [comment]
I never use that feature, but know others do.

I take it you use this in latex?

I know org-cite will also support local bibs.

@ilupin any input?

**Reply by bdarcus** [reply]
I didn't see this was a discussion thread, so created an issue for it.

https://github.com/bdarcus/bibtex-actions/issues/150

**Comment by ilupin** [comment]
A buffer can have a local `bibtex-completion-bibliography` by `M-x add-file/dir-local-variable`.

However, it seems `bibtex-actions` has a global `bibtex-actions--candidates-cache` which prevents `bibtex-actions-*` commands getting the latest candidates. You have to refresh the cache with `C-u M-x bibtex-action-*`.

**Reply by bdarcus** [reply]
@ramos can you let us know if this works?

If yeah there's a bit of a long story for why this doesn't auto-update the cache the same as ivy-bibtex and helm-bibtex, which I can go into ...

**Reply by bdarcus** [reply]
Here's the wiki section on this, which I just updated with a `watchexec` config example that should work.

https://github.com/bdarcus/bibtex-actions/wiki/Configuration#proactive-reloading-of-library
